### PR TITLE
Add new workflow to pin dependencies automatically and add the pinned files

### PR DESCRIPTION
**Pull request by mihaimaruseac**
The workflow needs to run as workflow dispatch after which it will upload the pinned dependencies as an artifact. A maintainer then will have to create a PR with the new ones. I didn't want to make the workflow write directly to the repo.

I also uploaded the generated requirement files and removed the `slsa_for_models/install/requirements_mac.txt` file which was not in use.

What is left is to make SLSA for models workflows also work on Windows and MacOS, in a future PR.

Signed-off-by: Mihai Maruseac <mihaimaruseac@google.com>

### CODE REVIEW

File: model_signing/install/requirements.in [hunk 1]
```diff
@@ -1,2 +1,2 @@
-psutil==5.9.5
-sigstore==2.1.0
+psutil
```
[the remark this answers is not in the document]
Comment (mihaimaruseac) [reply model_signing/install/requirements.in:1]: Sadly dependabot doesn't seem to upgrade `.in` files, so we would only use this workflow when we need to add a new dependency or do a manual upgrade. In both cases, having pins can just slow us down.

Dependabot only updates the `requirements*.txt` files, which also means that having bounds in `.in` when we run the workflow again we'll revert changes made by dependabot

[the remark this answers is not in the document]
Comment (laurentsimon) [reply model_signing/install/requirements.in:1]: Got it. So we need to run it on a schedule then? Weekly? I suppose they will start interfering with each other if we don't merge fast enough? Is it worth restricting to minor versions or not? 

Let's discuss in another issue then. Let's start with workflow_dispatch :)

**Review by laurentsimon — APPROVED** [review]
Thanks, this is great!

File: model_signing/install/requirements.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (mihaimaruseac) [reply model_signing/install/requirements.in:1]: Done in #99

**Pull request merged**